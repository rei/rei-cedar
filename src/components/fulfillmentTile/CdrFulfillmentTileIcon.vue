<script setup lang="ts">
import { useCssModule, computed } from 'vue';
import { fulfillmentTileIcon } from '../../types/interfaces';

/** Provides stylized icons for typical notifications */

defineOptions({ name: 'CdrFulfillmentTileIcon' });

const props = withDefaults(defineProps<fulfillmentTileIcon>(), {
  type: 'default',
});

const style = useCssModule();
const baseClass = 'cdr-fulfillment-tile-icon';
const typeClass = computed(() => `${baseClass}--${props.type}`);
</script>

<template>
  <span :class="[style[baseClass], style[typeClass]]">
    <!-- @slot Where all default content should be placed. -->
    <slot />
  </span>
</template>

<style lang="scss" module src="./styles/CdrFulfillmentTileIcon.module.scss"></style>
