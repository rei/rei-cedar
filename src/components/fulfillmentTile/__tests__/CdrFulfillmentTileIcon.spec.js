import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrFulfillmentTileIcon from '../CdrFulfillmentTileIcon.vue';

describe('CdrFulfillmentTileIcon', () => {
  it('default snapshot', () => {
    const wrapper = mount(CdrFulfillmentTileIcon);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('has default type class', () => {
    const wrapper = mount(CdrFulfillmentTileIcon);
    expect(wrapper.classes()).toContain('cdr-fulfillment-tile-icon');
    expect(wrapper.classes()).toContain('cdr-fulfillment-tile-icon--default');
    expect(wrapper.element).toMatchSnapshot();
  });

  it('applies info type class', () => {
    const wrapper = mount(CdrFulfillmentTileIcon, { props: { type: 'info' } });
    expect(wrapper.classes()).toContain('cdr-fulfillment-tile-icon--info');
    expect(wrapper.element).toMatchSnapshot();
  });

  it('applies warning type class', () => {
    const wrapper = mount(CdrFulfillmentTileIcon, { props: { type: 'warning' } });
    expect(wrapper.classes()).toContain('cdr-fulfillment-tile-icon--warning');
    expect(wrapper.element).toMatchSnapshot();
  });

  it('applies success type class', () => {
    const wrapper = mount(CdrFulfillmentTileIcon, { props: { type: 'success' } });
    expect(wrapper.classes()).toContain('cdr-fulfillment-tile-icon--success');
    expect(wrapper.element).toMatchSnapshot();
  });

  it('applies error type class', () => {
    const wrapper = mount(CdrFulfillmentTileIcon, { props: { type: 'error' } });
    expect(wrapper.classes()).toContain('cdr-fulfillment-tile-icon--error');
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders default slot content', () => {
    const wrapper = mount(CdrFulfillmentTileIcon, {
      slots: { default: '<svg data-testid="icon-svg"></svg>' }
    });
    expect(wrapper.html()).toContain('data-testid="icon-svg"');
    expect(wrapper.element).toMatchSnapshot();
  });
});