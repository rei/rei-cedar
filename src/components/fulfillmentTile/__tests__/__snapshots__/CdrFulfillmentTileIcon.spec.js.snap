// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrFulfillmentTileIcon > applies error type class 1`] = `
<span
  class="cdr-fulfillment-tile-icon cdr-fulfillment-tile-icon--error"
>
  <!-- @slot Where all default content should be placed. -->
  
  
</span>
`;

exports[`CdrFulfillmentTileIcon > applies info type class 1`] = `
<span
  class="cdr-fulfillment-tile-icon cdr-fulfillment-tile-icon--info"
>
  <!-- @slot Where all default content should be placed. -->
  
  
</span>
`;

exports[`CdrFulfillmentTileIcon > applies success type class 1`] = `
<span
  class="cdr-fulfillment-tile-icon cdr-fulfillment-tile-icon--success"
>
  <!-- @slot Where all default content should be placed. -->
  
  
</span>
`;

exports[`CdrFulfillmentTileIcon > applies warning type class 1`] = `
<span
  class="cdr-fulfillment-tile-icon cdr-fulfillment-tile-icon--warning"
>
  <!-- @slot Where all default content should be placed. -->
  
  
</span>
`;

exports[`CdrFulfillmentTileIcon > default snapshot 1`] = `
<span
  class="cdr-fulfillment-tile-icon cdr-fulfillment-tile-icon--default"
>
  <!-- @slot Where all default content should be placed. -->
  
  
</span>
`;

exports[`CdrFulfillmentTileIcon > has default type class 1`] = `
<span
  class="cdr-fulfillment-tile-icon cdr-fulfillment-tile-icon--default"
>
  <!-- @slot Where all default content should be placed. -->
  
  
</span>
`;

exports[`CdrFulfillmentTileIcon > renders default slot content 1`] = `
<span
  class="cdr-fulfillment-tile-icon cdr-fulfillment-tile-icon--default"
>
  <!-- @slot Where all default content should be placed. -->
  
  <svg
    data-testid="icon-svg"
  />
  
</span>
`;
